tests/unit/restapi/v1/test_tags.py

```diff
@@ -0,0 +1,413 @@
+        client: The Flask test client.
+        name: The name to assign to the new tag.
+
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = actions.register_tag(client, name=name)
+    assert response.status_code == 400
+
+
+def assert_tag_name_matches_expected_name(
+    client: FlaskClient, tag_id: int, expected_name: str
+) -> None:
+    """Assert that the name of a tag matches the expected name.
+
+    Args:
+        client: The Flask test client.
+        tag_id: The id of the tag to retrieve.
+        expected_name: The expected name of the tag.
+
+    Raises:
+        AssertionError: If the response status code is not 200 or if the name of the
+            tag does not match the expected name.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_TAGS_ROUTE}/{tag_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 200 and response.get_json()["name"] == expected_name
+
+
+def assert_tag_is_not_found(
+    client: FlaskClient,
+    tag_id: int,
+) -> None:
+    """Assert that a tag is not found.
+
+    Args:
+        client: The Flask test client.
+        tag_id: The id of the tag to retrieve.
+
+    Raises:
+        AssertionError: If the response status code is not 404.
+    """
+    response = client.get(
+        f"/{V1_ROOT}/{V1_TAGS_ROUTE}/{tag_id}",
+        follow_redirects=True,
+    )
+    assert response.status_code == 404
+
+
+def assert_cannot_rename_tag_with_existing_name(
+    client: FlaskClient,
+    tag_id: int,
+    existing_name: str,
+) -> None:
+    """Assert that renaming a tag with an existing name fails.
+
+    Args:
+        client: The Flask test client.
+        tag_id: The id of the tag to rename.
+        name: The name of an existing tag.
+
+    Raises:
+        AssertionError: If the response status code is not 400.
+    """
+    response = rename_tag(
+        client=client,
+        tag_id=tag_id,
+        new_name=existing_name,
+    )
+    assert response.status_code == 400
+
+
+# -- Tests -----------------------------------------------------------------------------
+
+
+@pytest.mark.v1
+def test_create_tag(
+    client: FlaskClient,
+    db: SQLAlchemy,
+    auth_account: dict[str, Any],
+) -> None:
+    """Test that tags can be registered and retrieved using the API.
+
+    This test validates the following sequence of actions:
+
+    - A user registers a tag: {name: "tag"}
+    - The user is able to retrieve information about the tag using the
+      tag id.
+    - The returned information matches the information that was provided
+      during registration.
+    """
+    tag_response = actions.register_tag(
+        client,
+        name="tag",
+        group_id=auth_account["default_group_id"],
+    )
+    tag_expected = tag_response.get_json()
+
+    assert_tag_response_contents_matches_expectations(
+        response=tag_expected,
+        expected_contents={
+            "name": tag_expected["name"],
```

Comment:
We don't want to use the output of registered tag here, since we want to ensure the response matches what the user input.

Instead create a variable `name="tag"` and pass it to both the register function and this assert.

Reply:
Added with commit: [58ffe1f](https://github.com/usnistgov/dioptra/pull/455/commits/58ffe1f8d4b851bc255d8a93e2608f3c331483c2)